src/DataTable/TableRow.js

```diff
@@ -70,6 +70,16 @@ class TableRow extends PureComponent {
     expanded: false,
   }
 
+  componentDidMount() {
```

Comment:
I just noticed this but I would suggest not calling setState on componentDidMount. Instead, just set a default state in the constructor.

I think you could even just change the this.state to

```
  state = {
    expanded: row[defaultExpandedField] || false,
  }
```

Reply:
The thing here is that defaultExpandedField comes from context, not props... And in the constructor method, context isn't available yet. That's the real reason for using setState in the componentDidMount.

But, if you know a better way to do this, please tell me how we can do it.

Reply:
I believe you can just do this.context. if that doesn't work just set things up in a `constructor`

```
  state = {
    expanded: this.props.row[this.context.defaultExpandedField] || false,
  }
```

Reply:
Or if this.state doesn't work for some reason:
```
constructor(props) {
  super (props);
  this.state = {
    expanded:props.row[this.context.defaultExpandedField] || false,
  }
}
```

Reply:
Ahh ok. I see. That's interesting that this.context is not available on the constructor

Reply:
I already tried that, but this.context in the constructor is undefined

Reply:
so you can access context as a parm on the constructur
```
  constructor(props, context) {
    super(props, context);

    this.state = {
      expanded: props.row[context.defaultExpandedField],
    };
  }
```

Reply:
What I think I can do is this in the renderRows method:

`this.calculateRows().map((row, i) => (
<TableRow
          key={row[keyField] || i}
          row={row}
          defaultExpanded={row[defaultExpandedField] || false}
          onRowClicked={this.handleRowClicked}
          onRowSelected={this.handleRowSelected}
        />
      ))`

Reply:
@jbetancur I did the change from my [last comment](https://github.com/jbetancur/react-data-table-component/pull/148#discussion_r303257228). I think this a better solution to avoid using setState in the componentDidMount. Please check it and tell me if it's okay.

Reply:
> so you can access context as a parm on the constructor
> 
> ```
>   constructor(props, context) {
>     super(props, context);
> 
>     this.state = {
>       expanded: props.row[context.defaultExpandedField],
>     };
>   }
> ```

I made a little digging and I found that this context param in the constructor method seems to be (or going to be) deprecated. Look at this [issue](https://github.com/facebook/react/issues/6598)

Reply:
Bummer! Although it feels weird to have to pass in props that are also available on the context.  But that is a design issue with TableRow; which doesn't really need access to context since it is 1 level deep. I'll fix that in the future